include/seqan/seq_io/bam_sam.h

```diff
@@ -0,0 +1,545 @@
+{
+    while (nextIs(file.iter, SamHeader()))
+    {
+        skipLine(file.iter);
+    }
+}
+// ----------------------------------------------------------------------------
+// Function skipHeader()                                              BamHeader
+// ----------------------------------------------------------------------------
+template <typename TFile>
+inline void skipHeader(TFile & file, Bam const & /* format */)
+{
+    BamHeader header;
+    BamFileIn bamIn;
+    open(bamIn, file.stream);
+    file.context.bamIOContext = context(bamIn);
+    readHeader(header, file.context.bamIOContext, file.iter, Bam());
+}
+
+// ----------------------------------------------------------------------------
+// Function readRecord                                                     Bam
+// ----------------------------------------------------------------------------
+template <typename TIdString, typename TSeqString, typename TCharIter>
+inline int32_t readBamRecord(TIdString & meta,
+                             TSeqString & seq,
+                             TIdString & prevQName,
+                             int32_t & remainingBytes,
+                             TCharIter & it)
+{
+    typedef typename Iterator<TSeqString, Standard>::Type             TSeqIter;
+
+    // BamAlignmentRecordCore.
+    BamAlignmentRecordCore recordCore;
+    arrayCopyForward(it, it + sizeof(BamAlignmentRecordCore), reinterpret_cast<char *>(&recordCore));
+    it += sizeof(BamAlignmentRecordCore);
+
+    clear(meta);
+    clear(seq);
+
+    remainingBytes -= sizeof(BamAlignmentRecordCore) + recordCore._l_qname +
+                      recordCore._n_cigar * 4 + (recordCore._l_qseq + 1) / 2 + recordCore._l_qseq;
+    SEQAN_ASSERT_GEQ(remainingBytes, 0);
+
+    // query name.
+    resize(meta, recordCore._l_qname - 1, Exact());
+    arrayCopyForward(it, it + recordCore._l_qname - 1, begin(meta, Standard()));
+    it += recordCore._l_qname;
+
+    if (prevQName == meta)
+        return 0;
+
+    // skip cigar string.
+    it += sizeof(uint32_t) * recordCore._n_cigar;
+
+    // query sequence.
+    resize(seq, recordCore._l_qseq, Exact());
+    TSeqIter sit = begin(seq, Standard());
+    TSeqIter sitEnd = sit + (recordCore._l_qseq & ~1);
+    while (sit != sitEnd)
+    {
+        unsigned char ui = getValue(it);
+        ++it;
+        assignValue(sit, Iupac(ui >> 4));
+        ++sit;
+        assignValue(sit, Iupac(ui & 0x0f));
+        ++sit;
+    }
+    if (recordCore._l_qseq & 1)
+        *sit++ = Iupac((uint8_t)*it++ >> 4);
+    return recordCore._l_qseq; // returns length of query sequence. Needed for reading quality
+}
+
+
+// ----------------------------------------------------------------------------
+// Function readRecord
+// ----------------------------------------------------------------------------
+
+template <typename TIdString, typename TSeqString, typename TForwardIter>
+inline bool readSamRecord(TIdString & meta,
```

Comment:
function name, see above
also concerning the linebreaks in a function header, it should be
```c++
template <typename T>
inline void functionName(T arg, T arg2)
...
```
Then if `inline void functionName()` doesn't fit on one line, split the return type into an extra line
```c++
template <typename T>
inline void 
functionName(T arg, T arg2)
...
```
and then if  `functionName()` doesn't fit into 120 characters, split at the arguments:
```c++
template <typename T>
inline void 
functionName(T arg,
             T arg2,
             T arg2,
             T arg2,
             T arg2,
             T arg2,
             T arg2,
             T arg2,
             T arg2,
             ...)
...
```
(in your case you split needlessly at the arguments but didnt split at the return type)